Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -507,6 +520,88 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::slice)
     return js_string(vm, string_part);
 }
 
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::split)
+{
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto* result = Array::create(global_object);
+    size_t result_len = 0;
+
+    size_t limit = NumericLimits<size_t>::max();
+    if (!vm.argument(1).is_undefined())
+        limit = vm.argument(1).to_size_t(global_object);
+
+    auto separator = vm.argument(0).to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    if (limit == 0)
+        return result;
+
+    if (vm.argument(0).is_undefined()) {
+        if (!result->define_property(0, js_string(vm, string))) {
+            vm.throw_exception<TypeError>(global_object, ErrorType::ObjectDefinePropertyReturnedFalse);
+            return {};
+        }
+        return result;
```

Comment:
You can skip this, we fully control the `result` array and there's no way `define_property()` will return false for this case :)

```suggestion
        result->define_property(0, js_string(vm, string));
        return result;
```

or, to be extra sure:

```suggestion
        auto success = result->define_property(0, js_string(vm, string));
        ASSERT(success);
        return result;
```

Reply:
It was like that before but I changed it to what  it is in the last minute :D